sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -1279,107 +1341,57 @@ static bool _resolve_addr(const ipv6_addr_t *dst, gnrc_netif_t *netif,
-            /* reduce back-off to possibly resolve neighbor sooner again */
-            entry->ns_sent = 3;
-        }
-#endif  /* CONFIG_GNRC_IPV6_NIB_ARSM */
-        if (pkt != NULL) {
-#if IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_QUEUE_PKT)
-            if (_get_nud_state(entry) == GNRC_IPV6_NIB_NC_INFO_NUD_STATE_INCOMPLETE) {
-                gnrc_pktqueue_t *queue_entry = _alloc_queue_entry(pkt);
-
-                if (queue_entry != NULL) {
-                    if (netif != NULL) {
-                        gnrc_pktsnip_t *netif_hdr = gnrc_netif_hdr_build(
-                                NULL, 0, NULL, 0
-                            );
-                        if (netif_hdr == NULL) {
-                            DEBUG("nib: can't allocate netif header for queue\n");
-                            gnrc_pktbuf_release(pkt);
-                            queue_entry->pkt = NULL;
-                            return false;
-                        }
-                        gnrc_netif_hdr_set_netif(netif_hdr->data, netif);
-                        queue_entry->pkt = gnrc_pkt_prepend(queue_entry->pkt,
-                                                            netif_hdr);
-                    }
-                    gnrc_pktqueue_add(&entry->pktqueue, queue_entry);
-                }
-                else {
-                    DEBUG("nib: can't allocate entry for packet queue "
-                          "dropping packet\n");
-                    gnrc_pktbuf_release(pkt);
-                    return false;
-                }
-            }
-            /* pkt != NULL already checked above */
-            else {
-                gnrc_icmpv6_error_dst_unr_send(ICMPV6_ERROR_DST_UNR_ADDR,
-                                               pkt);
-                gnrc_pktbuf_release_error(pkt, EHOSTUNREACH);
-            }
-#else   /* CONFIG_GNRC_IPV6_NIB_QUEUE_PKT */
-            gnrc_icmpv6_error_dst_unr_send(ICMPV6_ERROR_DST_UNR_ADDR,
-                                           pkt);
-            gnrc_pktbuf_release_error(pkt, EHOSTUNREACH);
-#endif  /* CONFIG_GNRC_IPV6_NIB_QUEUE_PKT */
+        if (IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_ROUTER) && netif != NULL) {
+            _call_route_info_cb(netif,
+                                GNRC_IPV6_NIB_ROUTE_INFO_TYPE_NSC,
+                                dst,
+                                (void *)GNRC_IPV6_NIB_NC_INFO_NUD_STATE_INCOMPLETE);
         }
+        reset = true;
+    }
 #if IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_ARSM)
+    else if (_get_nud_state(entry) == GNRC_IPV6_NIB_NC_INFO_NUD_STATE_UNREACHABLE) {
+        /* reduce back-off to possibly resolve neighbor sooner again */
+        entry->ns_sent = 3;
+    }
+#endif
+
+    /* queue packet as we have to do address resolution first */
+    if (pkt != NULL && !_enqueue_for_resolve(netif, pkt, entry)) {
+        return false;
+    }
+
+    if (IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_ARSM)) {
```

Comment:
I have a feeling that this should in fact be 
```suggestion
    if (IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_ARSM) && !gnrc_netif_is_6lo(netif)) {
```

Reply:
No, it shoud still be possible to activate the ARSM optionally on 6Lo nodes.

Reply:
(also resolution by address generation reversal is also a nice feature for non-6Lo-nodes to have (and will become necessary with SCHC at some point too).

Reply:
> No, it shoud still be possible to activate the ARSM optionally on 6Lo nodes.

But that would be a separate config then, right? Because right now we only have `CONFIG_GNRC_IPV6_NIB_ARSM` which globally turns on ARSM on *all* interfaces.

So the BR will do ARSM on the 6lo interface (because it needs it on the upstream interface) but the 6lo nodes do not.

Reply:
In my opinion, `CONFIG_GNRC_IPV6_NIB_ARSM` should even be enabled for all 6LN. Because our `CONFIG_GNRC_IPV6_NIB_ARSM` is actually responsible for two things:
- address resolution obviously
(Note: with private addresses, 6LN cannot reverse translate IPv6 addresses to L2 addresses, so they must look in the NC anyway. But this is a bit off-topic here currently.)
- and Neighbor Unreachable Detection

6LN actually do NUD [RFC6775](https://datatracker.ietf.org/doc/html/rfc6775#section-5.5).
```
Hosts send unicast NS messages to register their IPv6 addresses, and
also to do NUD to verify that their default routers are still
reachable.
```
For some things like joining the Solicited-node multicast address and multicast L2 address resolution, we just exclude 6LN interfaces.

Reply:
> Neighbor Unreachable Detection

hm might this be the reason for #17327

Reply:
very hard to tell... To be honest I am not very familiar with RPL. I have to catch up here.